src/category_theory/flat_functors.lean

```diff
@@ -0,0 +1,266 @@
+  (s : cone (K ⋙ F)) (f₁ f₂ : s.X ⟶ F.obj c.X)
+  (h₁ : ∀ (j : J), f₁ ≫ (F.map_cone c).π.app j = s.π.app j)
+  (h₂ : ∀ (j : J), f₂ ≫ (F.map_cone c).π.app j = s.π.app j) : f₁ = f₂ :=
+begin
+  -- We can make two cones over the diagram of `s` via `f₁` and `f₂`.
+  let c₁ : cone (to_diagram s ⋙ pre s.X K F) := (cones.postcompose
+    (eq_to_hom (by simpa [←to_diagram_comp_map s (F.map_cone c) f₁ h₁]))).obj (to_cone F c f₁),
+  let c₂ : cone (to_diagram s ⋙ pre s.X K F) := (cones.postcompose
+    (eq_to_hom (by simpa [←to_diagram_comp_map s (F.map_cone c) f₂ h₂]))).obj (to_cone F c f₂),
+
+  -- The two cones can then be combined and we may obtain a cone over the two cones since
+  -- `structured_arrow s.X F` is cofiltered.
+  let c₀ := is_cofiltered.cone (bicone_mk _ c₁ c₂),
+  let g₁ : c₀.X ⟶ c₁.X := c₀.π.app (bicone.left),
+  let g₂ : c₀.X ⟶ c₂.X := c₀.π.app (bicone.right),
+
+  -- Then `g₁.right` and `g₂.right` are two maps from the same cone into the `c`.
+  have : ∀ (j : J), g₁.right ≫ c.π.app j = g₂.right ≫ c.π.app j,
+  { intro j,
+    injection c₀.π.naturality (bicone_hom.left  j) with _ e₁,
+    injection c₀.π.naturality (bicone_hom.right j) with _ e₂,
+    simpa [eq_to_hom_right] using e₁.symm.trans e₂ },
+  have : c.extend g₁.right = c.extend g₂.right,
+  { unfold cone.extend, congr' 1, ext x, apply this },
+
+  -- And thus they are equal as `c` is the limit.
+  have : g₁.right = g₂.right,
+  calc g₁.right = hc.lift (c.extend g₁.right) : by { apply hc.uniq (c.extend _), tidy }
+            ... = hc.lift (c.extend g₂.right) : by { congr, exact this }
+            ... = g₂.right                    : by { symmetry, apply hc.uniq (c.extend _), tidy },
+
+  -- Finally, since `fᵢ` factors through `F(gᵢ)`, the result follows.
+  calc f₁ = 𝟙 _ ≫ f₁                  : by simp
+      ... = c₀.X.hom ≫ F.map g₁.right : g₁.w
+      ... = c₀.X.hom ≫ F.map g₂.right : by rw this
+      ... = 𝟙 _ ≫ f₂                  : g₂.w.symm
+      ... = f₂                         : by simp
+end
+
+end preserves_finite_limit_of_flat
+
+/-- Representably flat functors preserves finite limits. -/
+noncomputable
+def preserves_finite_limit_of_flat (F : C ⥤ D) [representably_flat F]
+(J : Type v₁) [small_category J] [fin_category J]
+: preserves_limits_of_shape J F := ⟨λ K, ⟨λ c hc,
+{ lift := preserves_finite_limit_of_flat.lift F hc,
+  fac' := preserves_finite_limit_of_flat.fac F hc,
+  uniq' := λ s m h, by
+  { apply preserves_finite_limit_of_flat.uniq F hc,
+    exact h,
+    exact preserves_finite_limit_of_flat.fac F hc s } }⟩⟩
+
+end has_limit
+
+section small_category
+variables {C D : Type u₁} [small_category C] [small_category D]
+
+/--
+(Implementation)
+The evaluation of `Lan F` at `X` is the colimit over the costrucuted arrows over `X`.
+-/
+lemma Lan_evaluation_eq_colim (E : Type u₂) [category.{u₁} E] (F : C ⥤ D) (X : D)
+  [∀ (X : D), has_colimits_of_shape (costructured_arrow F X) E] :
+  Lan F ⋙ (evaluation D E).obj X =
```

Comment:
Equality of functors is typically a bad sign. In your case you use it in `Lan_preserves_finite_limit_of_flat`, where it would be okay to use isomorphism of functors instead (which is safe to use), and preserving limits is preserved by nat isomorphism in the functor.